R/validate_vpts.R

```diff
@@ -0,0 +1,107 @@
+vpts_schema <- get0("vpts_schema", envir = asNamespace("bioRad"))
+
+get_field_schema <- function(field, schema) {
+    for (i in seq_along(schema$fields$name)) {
+        if (field %in% c(schema$fields$name[i], schema$fields$nameAlternatives[[i]])) {
+            return(schema$fields[i, ])
+        }
+    }
+    return(NULL)  # return NULL if no matching field found
+}
+validate_datetime_format <- function(data, format) {
+  parsed_data <- tryCatch({
+    as.POSIXct(data, format = format, tz = "UTC")
+  }, error = function(e) NULL)
+  # check for failed parsing
+  if (any(is.na(parsed_data))) {
+    return(FALSE)
+  }
+  return(TRUE)
+}
+#' Validate dataframe against VPTS schema
+#'
+#' @param df The dataframe to validate.
+#'
+#' @return Invisibly returns a list with validation results including messages for any issues found.
+#' @importFrom glue glue
```

Comment:
All @importFrom statements can be removed, as long as we use `package::function` format to refer to external functions. This applies also to the following files:

```
as.vpts.R:#' @importFrom glue glue
map.R:#' @importFrom methods as
read_vpts.R:#' @importFrom readr read_csv
utils.R:#' @importFrom stats na.omit
validate_vpts.R:#' @importFrom glue glue
validate_vpts.R:#' @importFrom dplyr select
validate_vpts.R:#' @importFrom stringr str_detect
```